src/api/alchemy.ts

```diff
@@ -507,9 +525,395 @@ export class Alchemy {
+   * been mined.
+   *
+   * To stall until the transaction has been mined, consider the
+   * waitForTransaction method below.
+   *
+   * @param transactionHash The hash of the transaction to get.
+   * @public
+   */
+  async getTransactionReceipt(
+    transactionHash: string | Promise<string>
+  ): Promise<TransactionReceipt | null> {
+    const provider = await this.getProvider();
+    return provider.getTransactionReceipt(transactionHash);
+  }
+
+  /**
+   * Submits transaction to the network to be mined. The transaction must be
+   * signed, and be valid (i.e. the nonce is correct and the account has
+   * sufficient balance to pay for the transaction).
+   *
+   * @param signedTransaction The signed transaction to send.
+   * @public
+   */
+  async sendTransaction(
+    signedTransaction: string | Promise<string>
+  ): Promise<TransactionResponse> {
+    const provider = await this.getProvider();
+    return provider.sendTransaction(signedTransaction);
+  }
+
+  /**
+   * Returns a promise which will not resolve until specified transaction hash is mined.
+   *
+   * If {@link confirmations} is 0, this method is non-blocking and if the
+   * transaction has not been mined returns null. Otherwise, this method will
+   * block until the transaction has confirmed blocks mined on top of the block
+   * in which it was mined.
+   *
+   * @param transactionHash The hash of the transaction to wait for.
+   * @param confirmations The number of blocks to wait for.
+   * @param timeout The maximum time to wait for the transaction to confirm.
+   * @public
+   */
+  async waitForTransaction(
+    transactionHash: string,
+    confirmations?: number,
+    timeout?: number
+  ): Promise<TransactionReceipt | null> {
+    const provider = await this.getProvider();
+    return provider.waitForTransaction(transactionHash, confirmations, timeout);
+  }
+
+  /**
+   * Returns an array of logs that match the provided filter.
+   *
+   * @param filter The filter object to use.
+   * @public
+   */
+  async getLogs(
+    filter: Filter | FilterByBlockHash | Promise<Filter | FilterByBlockHash>
+  ): Promise<Array<Log>> {
+    const provider = await this.getProvider();
+    return provider.getLogs(filter);
+  }
+
+  /**
+   * Allows sending a raw message to the Alchemy backend.
+   *
+   * @param method The method to call.
+   * @param params The parameters to pass to the method.
+   * @public
+   */
+  async send(method: string, params: Array<any>): Promise<any> {
+    const provider = await this.getProvider();
+    return provider.send(method, params);
+  }
+
+  /**
+   * Adds a listener to be triggered for each {@link eventName} event. Also
+   * includes Alchemy's Subscription API events. See {@link AlchemyEventType} for
+   * how to use them.
+   *
+   * @param eventName The event to listen for.
+   * @param listener The listener to call when the event is triggered.
+   * @public
+   */
+  async on(
```

Comment:
One problem we now run into with dynamic imports is that we can't maintain the same API surface of `on(): AlchemyWebSocketProvider` anymore. I don't really see developers chaining these calls like `alchemy.on().off()` since most of these are meant to be one-off methods.

I'm okaaaay-ish with this based on the use case, but I wonder if you see any way to make this nicer @dphilipson. Thanks!

Reply:
Isn't the original behavior to return `this`? What's stopping us from doing that? E.g.
```ts
  on(
    eventName: AlchemyEventType,
    listener: Listener
  ): this {
    (async () => {
      const provider = await this.getWebsocketProvider();
      provider.on(eventName, listener);
    })();
    return this;
  }
```

Reply:
> What's stopping us from doing that? E.g.
Doh... only my ignorance. In my mind I thought needed to return the underlying provider rather than the top level. thanks!